### PR TITLE
First step 

### PR DESCRIPTION
**Pull request by wyaming89**
Add a Transfer entity

### CODE REVIEW

**Review by greefea — COMMENTED** [review]
Create three entities: 

token: id, symbol, supply, decimals
transfer: id (=event.transaction.hash), from, to, amount
dailyTransfer: id (day), transfers (number of transfers), amount (total amount transferred)

The id of dailyTransfer: 
```
let id = (event.block.timestamp.toI64() / SECONDS_PER_DAY).toString();
```

**Comment by wyaming89** [comment]
I would like to use natural dates, maybe for time series analysis……

`let day =  (event.block.timestamp.toI64() - (event.block.timestamp.toI64()%SECONDS_PER_DAY)).toString()`

**Comment by greefea** [comment]
Good progress! 

> I would like to use natural dates, maybe for time series analysis……
> 
> `let day = (event.block.timestamp.toI64() - (event.block.timestamp.toI64()%SECONDS_PER_DAY)).toString()`

Try: 

```

  let date = new Date(event.block.timestamp.toI32() * 1000)
  let day = date.toISOString().slice(0, 10);
```

File: src/dai.ts [hunk 1]
```diff
@@ -26,4 +40,17 @@ export function handleTransfer(event: Transfer): void {
...
+  if (transferArray.indexOf(event.params.src.toHexString(),0)==-1){
+    transferArray.push(event.params.src.toHexString());
+    daily.transfers = daily.transfers + 1;
+  }
+  if (transferArray.indexOf(event.params.dst.toHexString(),0)==-1){
+    transferArray.push(event.params.dst.toHexString());
+    daily.transfers = daily.transfers + 1;
+  }
+
+  daily.amount = daily.amount.plus(event.params.wad);
```
Comment (greefea) [inline src/dai.ts]: daiy.transfers = daiy.transfers + 1

**Comment by greefea** [comment]
For daily unique users, add a uniqueUsers entity with id = `${usr}`, where usr is src or dst. You can check whether a user is unique by checking if `uniqueUsers.load(id) == null`

**Comment by greefea** [comment]
DAI's price can be read from Chainlink's oracle contract (by google DAI chiainlink): https://etherscan.io/address/0xaed0c38402a5d19df6e4c03f4e2dced6e29c1ee9#readContract#F8. These are the steps you need to get DAI price in your handler function:
1. add an ABI file for chainlink contract to your subgraph
2. add the abi information to you subgraph.yaml
3. in your mapping.ts
    1. import the chainlink oracle contract
    2. instantiate it with `let contract = EACAggregatorProxy.bind(Address.fromString("0xAed0c38402a5d19df6E4c03F4E2DceD6e29c1ee9"))`
    3. `let price = contract.latestAnswer()` (pay attention to number of decimals)
4. multiple DAI amount with price to dollar amount

**Comment by wyaming89** [comment]
The first step is push, what will I learn next?

**Comment by greefea** [comment]
Good job! Have you deployed it to https://thegraph.com/hosted-service ? Can you provide a link to the deployed subgraph?

I edited README with tasks for second step.

**Comment by wyaming89** [comment]
> Good job! Have you deployed it to https://thegraph.com/hosted-service ? Can you provide a link to the deployed subgraph?
> 
> I edited README with tasks for second step.

I have deployed it to the graph studio. Does  "hosted-service" need to be learned?

**Comment by wyaming89** [comment]
> Good job! Have you deployed it to https://thegraph.com/hosted-service ? Can you provide a link to the deployed subgraph?
> 
> I edited README with tasks for second step.

subgraph link [https://thegraph.com/hosted-service/subgraph/wyaming89/dai_subgraph_001](url)

**Pull request merged**